src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -328,10 +328,11 @@
                                      ;; anything referencing an env or resource
                                      ;; param is purged first
                                      (garbage-collect! write-db)))]
-        ;; Run database maintenance tasks seqentially to avoid
-        ;; competition. Each task must handle its own errors.
-        (gc-task db-maintenance-tasks)
-        (gc-task #(compress-dlo! dlo-compression-threshold discard-dir)))
+        (when (pos? gc-interval-millis)
+          ;; Run database maintenance tasks seqentially to avoid
+          ;; competition. Each task must handle its own errors.
+          (gc-task db-maintenance-tasks)
+          (gc-task #(compress-dlo! dlo-compression-threshold discard-dir))))
```

Comment:
this raises another question -- under the scenario I described above the DLO dirs that matter will be on the command processing slaves, despite the fact that the database we care about GCing is on a different host. Existing issue, but it seems like we need to decouple the DLO compression from the database cleanup.